FormalConjectures/ErdosProblems/416.lean

```diff
@@ -0,0 +1,94 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 416
+
+*Reference:* [erdosproblems.com/416](https://www.erdosproblems.com/416)
+-/
+
+open Classical Filter
+open scoped Topology Real
+
+/--Let `V(x)` count the number of `n≤x` such that `ϕ(m)=n` is solvable-/
+noncomputable abbrev V (x : ℝ) : ℝ :=
+  (Finset.Icc 1 ⌊x⌋₊ |>.filter (fun n => ∃ (m : ℕ), m.totient = n)).card
+
+/--
+Let `V(x)` count the number of `n≤x` such that `ϕ(m)=n` is solvable. Does `V(2x)/V(x)→2` ?
+-/
+@[category research open, AMS 11]
+theorem erdos_416.parts.i :
+    Filter.Tendsto (fun x => (V (2 * x) / V (x))) Filter.atTop (𝓝 2) := by
+  sorry
+
+/--
+Let `V(x)` count the number of `n≤x` such that `ϕ(m)=n` is solvable.
+Is there an asymptotic formula for `V(x)`?
+-/
+@[category research open, AMS 11]
+theorem erdos_416.parts.ii :
+    let f : ℝ → ℝ := answer(sorry)
+    Filter.Tendsto (fun x => V x / f x) atTop (𝓝 1) := by
+  sorry
+
+/--
+Let `V(x)` count the number of `n≤x` such that `ϕ(m)=n` is solvable.
+Pillai proved `V(x)=o(x)`.
+Ref: S. Sivasankaranarayana, _On some functions connected with {$\phi(n)$_
+-/
+@[category research solved, AMS 11]
+theorem erdos_416.variants.Pillai : V =o[atTop] id := by
+  sorry
+
+/--
+Let `V(x)` count the number of `n≤x` such that `ϕ(m)=n` is solvable.
+Erdős proved V(x)=x(logx)^(−1+o(1)).
+Ref: Erdős, P., _On the normal number of prime factors of $p-1$ and some related problems concerning Euler's $\varphi$-function._
+-/
+@[category research solved, AMS 11]
+theorem erdos_416.variants.Erdos : ∃ f : ℝ → ℝ, f =o[atTop] (1 : ℝ → ℝ) ∧
+    ∀ᶠ x in Filter.atTop, V x = x * x.log ^ (-1 + f x) := by
+  sorry
+
+/--
+Let `V(x)` count the number of `n≤x` such that `ϕ(m)=n` is solvable.
+`V(x)=x/logx * e^((C+o(1))(log log log x)^2)`, for some explicit constant `C>0`.
+Ref:Maier, Helmut and Pomerance, Carl, _On the number of distinct values of Euler's $\phi$-function_.
+-/
+@[category research solved, AMS 11]
+theorem erdos_416.variants.Maier_Pomerance :
+    let C : ℝ := answer(sorry)
+    0 < C ∧ ∃ f : ℝ → ℝ, f =o[atTop] (1 : ℝ → ℝ) ∧
+      ∀ᶠ x in Filter.atTop, (V x : ℝ) = x / x.log * (rexp <| (C + f x) * x.log.log.log ^ 2) := by
+  sorry
+
+/--
+Let `V(x)` count the number of `n≤x` such that `ϕ(m)=n` is solvable.
+`V(x) ≍ x/log x*e^(C_1*(log log log x − log log log log x)^2+C_2 log log log x − C_3 log log log log x)`
+Ref: Ford, Kevin, _The distribution of totients_.
+-/
+@[category research solved, AMS 11]
+theorem erdos_416.variants.Ford :
+    let (C₁, C₂, C₃) : ℝ × ℝ × ℝ := answer(sorry)
+    0 < C₁ ∧ 0 < C₂ ∧ 0 < C₃ ∧
+    ∃ (f : ℝ → ℝ) (C : ℝ), (∀ᶠ x in Filter.atTop, |f x| ≤ C) ∧
+    let G (x : ℝ) : ℝ := x / x.log* (rexp <| C₁ * (x.log.log.log - x.log.log.log.log) ^ 2
+        + C₂* x.log.log.log - C₃ * x.log.log.log.log + f x)
+    V =Θ[atTop] G := by
```

Comment:
is the `f : ℝ → ℝ` necessary here, or would it be equivalent to drop it, since it would be swallowed by the `Θ` anyhow? (Not entirely sure..)